src/lib/desmoContract-module.ts

```diff
@@ -88,41 +88,48 @@ export class DesmoContract {
   }
 
   private async fetchAppOrder(appAddress: string): Promise<AppOrder> {
+    if (this.iexec === undefined) {
+      throw Error('A connection to iExec is required!');
+    }
     const { orders: appOrders } = await this.iexec.orderbook.fetchAppOrderbook(
       appAddress,
     );
 
-    const appOrder = appOrders && appOrders[0] && appOrders[0].order;
-
-    if (!appOrder) {
-      throw Error(`no apporder found for app ${appAddress}`);
+    if (appOrders.length > 0) {
+      return appOrders[0].order as AppOrder;

```

Comment:
Not related to these changes, but I was a little bit puzzled when I read the code. Why are we taking the first order in the array? What will happen if two clients use the SDK at the same time?

Reply:
Requiring help from @iotondato ...